src/device.cpp

```diff
@@ -29,6 +29,56 @@ cvk_device* cvk_device::create(cvk_platform* platform, VkInstance instance,
     return device;
 }
 
+void cvk_device::init_properties(VkInstance instance) {
+
+    cvk_info("Initialising properties");
+
+    // Get physical device properties
+    VkPhysicalDeviceProperties2KHR properties;
+    properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2_KHR;
+    properties.pNext = &m_driver_properties;
```

Comment:
Doing this unconditionally assumes that the extension is available or the version at least 1.2. This has been a problem for some people in the past in feature initialisation (see `cvk_device::init_features`). Given that device properties are necessary for extension initialisation, I think we'll have to query them twice (I can't think of a better way right now at least).

Reply:
A driver that doesn't support the extension is required to just skip the structure. From the [Vulkan spec](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#fundamentals-validusage-pNext):
> Any component of the implementation (the loader, any enabled layers, and drivers) must skip over, without processing (other than reading the sType and pNext members) any extending structures in the chain not defined by core versions or extensions supported by that component.

Are you saying that some Vulkan implementations don't actually do this? I tried to make the rest of the function robust to the extension not being supported (e.g. the `if (m_driver_properties.driverID != 0)` check). I'm happy to add a query for the extension if this is a real problem though.

Reply:
True. Yeah, it seems some implementations don't do this correctly. Let's try to keep it simple first as what you've done ought to work on a correct Vulkan implementation.

Reply:
Actually I've decided to do the extension check after all. It's useful to report when the extension is not present so we know when these behaviors are not being set for a device. Querying properties should be fast anyway.